index.js

```diff
@@ -122,21 +125,37 @@ Liftoff.prototype.buildEnvironment = function (opts) {
   }
 
   // use rechoir to autoload any required modules
-  var autoloads;
-  if (configPath) {
-    autoloads = rechoir.prepare(this.extensions, configPath, cwd, true);
-    if (autoloads instanceof Error) {
-      autoloads = autoloads.failures;
-    }
-    if (Array.isArray(autoloads)) {
-      autoloads.forEach(function (attempt) {
-        if (attempt.error) {
-          this.emit('requireFail', attempt.moduleName, attempt.error);
-        } else {
-          this.emit('require', attempt.moduleName, attempt.module);
-        }
-      }, this);
+  var registerLoader = function(extensions, configPath) {
+    var autoloads;
+    if (configPath) {
+      autoloads = rechoir.prepare(extensions, configPath, cwd, true);
+      if (autoloads instanceof Error) {
+        autoloads = autoloads.failures;
+      }
+      if (Array.isArray(autoloads)) {
+        autoloads.forEach(function (attempt) {
+          if (attempt.error) {
+            this.emit('requireFail', attempt.moduleName, attempt.error);
+          } else {
+            this.emit('require', attempt.moduleName, attempt.module);
+          }
+        }, this);
+      }
     }
+  }.bind(this);
+
+  registerLoader(this.extensions, configPath);
+
+  var configFiles = {};
+  if (isPlainObject(this.configFiles)) {
+    var fined = new Fined(cwd, this.extensions);
```

Comment:
@phated @tkellen I got it. I'll modify to using a pure function.

I'm considering the interface of this function. The parameters which `fined` has to receive are:

   * **pathObj** : a path object or a path string, and if it is an object the properties are:
       * a path string
       * a find-up flag
       * extensions, which can be a string, an array or an object.
   * **name** : a base name of searched file. (`configFiles`'s property key in `liftoff`)
   * **cwd** : a default base directory path.
   * **exts** : a default extensions, which can be a string, an array or an object.

And about the interface,

1. the most simple interface is:

   ```js
   var pathObj = { path: (mandatory), findUp: (optional) , extensions: (optional) };
   var file = fined(pathObj, name, cwd, exts); // `name`, `cwd`, `exts` are optional.
   ```

   But I think it would be confusing about these order for consumers.

2. the interface now I'm thinking are:

   ```js
   var pathObj = { path: (optional), findUp: (optional), extensions: (optional), cwd: (optional), name: (optional) };
   var defaultObj =  { path: (optional), findUp: (optional), extensions: (optional), cwd: (optional), name: (optional) };

   var file = fined(pathObj, defaultObj)
  ```

   `pathObj` and `defaultObj` can have same properties and lacking properties in `pathObj` are complemented with properties in `defaultObj` (by using `lodash.defaults`).

What do you think?